app/models/channels.js

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export class Channels {
+    static schema = {
+        name: 'Channels',
+        primaryKey: 'id',
+        properties: {
+            id: 'string',
+            createAt: 'int',
+            updateAt: 'int',
+            deleteAt: {type: 'int', default: 0},
+            team: 'Teams',
+            type: {type: 'string', indexed: true},
+            displayName: 'string?',
+            name: 'string',
+            header: 'string?',
+            lastPostAt: {type: 'int', default: 0},
+            totalMsgCount: {type: 'int', default: 0},
+            purpose: 'string?',
+            groupConstrained: {type: 'bool', default: false},
+            members: 'ChannelMembers[]',
+            memberCount: {type: 'int', default: 0},
+        },
+    }
+}
+
+export class ChannelMembers {
+    get channel() {
+        if (this.channel?.length) {
```

Comment:
Shouldn't these lines use `this.channels` instead of `this.channel`? As it is, I think it'll infinitely recurse